### PR TITLE
RCJ-30 - Integrating drift into codebase

### PR DESCRIPTION
**Pull request by NicholasDawson**
## Description

Mainly I was recreating all of our existing models as drift tables.

- I tested basic crud into the tables
- Imported all libraries for drift
- Created tables and enums for all necessary data at this point
- Standardized how we create models and wrote some docs for it [here](https://nmd2117.atlassian.net/wiki/spaces/RCJ/pages/1671169/Drift)
- Initialized data for tables that need it

Important Note: Our app is still not using drift for anything yet, this is an intermediate step before migrating objects in the app over to using our new drift objects.

## Ticket

RCJ-30 - [LINK](https://nmd2117.atlassian.net/browse/RCJ-30)

### CODE REVIEW

File: lib/database/models/climb_type/climb_type.dart [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  Future<int> add(ClimbTypeCompanion entry) {
+    return into(climbType).insert(entry);
+  }
+}
+
+enum ClimbTypeEnum {
+  topRope(1, "Top Rope", [GradeSystemEnum.yds]),
+  boulder(2, "Boulder", [GradeSystemEnum.vScale]),
+  lead(3, "Lead", [GradeSystemEnum.yds]);
```
Comment (devkav) [inline lib/database/models/climb_type/climb_type.dart:31]: Is this extensible in the future? Can other grade systems be used?

File: lib/database/models/tag/route_tag.dart [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import 'package:drift/drift.dart';
```
Comment (devkav) [inline lib/database/models/tag/route_tag.dart:1]: Does this class act as a connector that associates routes to tags?

**Review by devkav — COMMENTED** [review]
Is there anyway to convert from a database object directly to a dart object?

**Comment by NicholasDawson** [comment]
> Is there anyway to convert from a database object directly to a dart object?

There isn't a way to convert, but there doesn't need to be. The [Data class](https://nmd2117.atlassian.net/wiki/spaces/RCJ/pages/1671169/Drift?parentProduct=JSW&locale=en-US#Data-Object-Format%3A-%7BtableName%7DData) should be used within dart as the object.

File: lib/database/models/climb_type/climb_type.dart (hunk 1 above)
Comment (NicholasDawson) [reply lib/database/models/climb_type/climb_type.dart:31]: Yes, I made the third property `final List<GradeSystemEnum> validGradeSystems;` a list so more valid grade systems could be added to the same type.

File: lib/database/models/tag/route_tag.dart (hunk 2 above)
Comment (NicholasDawson) [reply lib/database/models/tag/route_tag.dart:1]: Yeah basically instead of storing a list in the db we want to do a separate table.

**Pull request merged**